new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java

```diff
@@ -0,0 +1,625 @@
+    }
+
+    String clusterService = domainUid + "-cluster-" + clusterName + "." + domainNamespace + ".svc.cluster.local";
+
+    Map<String, String> templateMap  = new HashMap();
+    templateMap.put("NAMESPACE", domainNamespace);
+    templateMap.put("DUID", domainUid);
+    templateMap.put("ADMIN_SERVICE",adminServerPodName);
+    templateMap.put("CLUSTER_SERVICE", clusterService);
+
+    Path srcHttpFile = Paths.get(RESOURCE_DIR, "istio", "istio-http-template.yaml");
+    Path targetHttpFile = assertDoesNotThrow(
+        () -> generateFileFromTemplate(srcHttpFile.toString(), "istio-http.yaml", templateMap));
+    logger.info("Generated Http VS/Gateway file path is {0}", targetHttpFile);
+
+    boolean deployRes = assertDoesNotThrow(
+        () -> deployHttpIstioGatewayAndVirtualservice(targetHttpFile));
+    assertTrue(deployRes, "Failed to deploy Http Istio Gateway/VirtualService");
+
+    int istioIngressPort = getIstioHttpIngressPort();
+    logger.info("Istio http ingress Port is {0}", istioIngressPort);
+
+    String consoleUrl = "http://" + K8S_NODEPORT_HOST + ":" + istioIngressPort + "/console/login/LoginForm.jsp";
+    boolean checkConsole =
+         checkAppUsingHostHeader(consoleUrl, domainNamespace + ".org");
+    assertTrue(checkConsole, "Failed to access WebLogic console");
+    logger.info("WebLogic console is accessible");
+
+    Path archivePath = Paths.get(ITTESTS_DIR, "../src/integration-tests/apps/testwebapp.war");
+    ExecResult result = null;
+    result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST, 
+        String.valueOf(istioIngressPort),
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT, 
+        clusterName, archivePath, domainNamespace + ".org", "testwebapp");
+    assertNotNull(result, "Application deployment failed");
+    logger.info("Application deployment returned {0}", result.toString());
+    assertEquals("202", result.stdout(), "Application deployed successfully");
+
+    String url = "http://" + K8S_NODEPORT_HOST + ":" + istioIngressPort + "/testwebapp/index.jsp";
+    logger.info("Application Access URL {0}", url);
+    boolean checkApp = checkAppUsingHostHeader(url, domainNamespace + ".org");
+    assertTrue(checkApp, "Failed to access WebLogic application");
+
+  }
+
+  /**
+   * Create a WebLogic domain on a persistent volume by doing the following.
+   * Create a configmap containing WLST script and property file.
+   * Create a Kubernetes job to create domain on persistent volume.
+   *
+   * @param wlstScriptFile       python script to create domain
+   * @param domainPropertiesFile properties file containing domain configuration
+   * @param pvName               name of the persistent volume to create domain in
+   * @param pvcName              name of the persistent volume claim
+   * @param namespace            name of the domain namespace in which the job is created
+   */
+  private void createDomainOnPVUsingWlst(Path wlstScriptFile, Path domainPropertiesFile,
```

Comment:
There is lot of duplicated code from ItDomainInPV which can be moved to common or util class.

Reply:
@anpanigr how about this ^?

Reply:
There are multiple such common functions which is repeated across many classes
 e.g.  createConfigMapForDomainCreation 
         createDomainOnPVUsingWlst
         createDomainJob
         createPV
         createPVC

Reply:
yes, they all need to be moved to a util class, its lot of duplicate code. If you don't want to do as part of this PR, create a JIRA and it should be prioritized.

Reply:
Maggie has already has created method for createPV and createPVC and createConfigMapForDomainCreation in CommonTestUtils class.  I am trying to use the method from CommonTestUtils.  However   createDomainOnPVUsingWlst() is a more involved one, it calls createDomainJob() with another set of parameters, so we have to change the signature for createDomainOnPVUsingWlst() with more parameters.

Reply:
Updated the ItIstioDomainInPV.java to use utility methods from utils/CommonTestUtils class
Add a JIRA to update other DomainInPV classes to do the same
https://jira.oraclecorp.com/jira/browse/OWLS-79135